lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -301,4 +301,334 @@ protected function get_block_classes( $style_nodes ) {
+			'color'  => null,
+			'radius' => null,
+			'style'  => null,
+			'width'  => null,
+		),
+		'color'           => array(
+			'background'       => null,
+			'custom'           => null,
+			'customDuotone'    => null,
+			'customGradient'   => null,
+			'defaultGradients' => null,
+			'defaultPalette'   => null,
+			'duotone'          => null,
+			'gradients'        => null,
+			'link'             => null,
+			'palette'          => null,
+			'text'             => null,
+		),
+		'custom'          => null,
+		'layout'          => array(
+			'contentSize' => null,
+			'wideSize'    => null,
+		),
+		'spacing'         => array(
+			'customSpacingSize' => null,
+			'spacingSizes'      => null,
+			'spacingScale'      => null,
+			'blockGap'          => null,
+			'margin'            => null,
+			'padding'           => null,
+			'units'             => null,
+		),
+		'typography'      => array(
+			'customFontSize' => null,
+			'dropCap'        => null,
+			'fontFamilies'   => null,
+			'fontSizes'      => null,
+			'fontStyle'      => null,
+			'fontWeight'     => null,
+			'letterSpacing'  => null,
+			'lineHeight'     => null,
+			'textDecoration' => null,
+			'textTransform'  => null,
+		),
+	);
+
+	/**
+	 * Given the block settings, it extracts the CSS Custom Properties
+	 * for the presets and adds them to the $declarations array
+	 * following the format:
+	 *
+	 * ```php
+	 * array(
+	 *   'name'  => 'property_name',
+	 *   'value' => 'property_value,
+	 * )
+	 * ```
+	 *
+	 * @param array $settings Settings to process.
+	 * @param array $origins  List of origins to process.
+	 * @return array Returns the modified $declarations.
+	 */
+	protected static function compute_preset_vars( $settings, $origins ) {
+		$declarations = array();
+
+		foreach ( static::PRESETS_METADATA as $preset_metadata ) {
+			$values_by_slug = static::get_settings_values_by_slug( $settings, $preset_metadata, $origins );
+
+			foreach ( $values_by_slug as $slug => $value ) {
+				$declarations[] = array(
+					'name'  => static::replace_slug_in_string( $preset_metadata['css_vars'], $slug ),
+					'value' => $value,
+				);
+			}
+		}
+
+		return $declarations;
+	}
+
+	/**
+	 * Transform the spacing scale values into an array of spacing scale presets.
+	 *
+	 * @param array $spacing_scale scale Array of values used to create the scale of spacing presets.
+	 * @return array An array of spacing preset sizes.
+	 */
+	protected static function get_spacing_sizes( $spacing_scale ) {
+		$spacing_sizes   = array();
+		$spacing_sizes[] = array(
+			'name' => 0,
+			'slug' => 0,
+			'size' => 0,
+		);
+		$spacing_sizes[] = array(
+			'name' => 1,
+			'slug' => 10,
+			'size' => $spacing_scale['firstStep'] . $spacing_scale['units'],
+		);
+
+		$current_step = $spacing_scale['firstStep'];
+
+		for ( $x = 1; $x <= $spacing_scale['steps'] - 1; $x++ ) {
+			$current_step = '+' === $spacing_scale['operator']
+				? $current_step + $spacing_scale['increment'] . $spacing_scale['units']
```

Comment:
This line's throwing a PHP notice for me:

<img width="1131" alt="image" src="https://user-images.githubusercontent.com/14988353/173717406-92a1a23c-aac4-48cf-97ae-327eb1c99086.png">

Reply:
Should be fixed now.